create_dataframes.py

```diff
@@ -0,0 +1,96 @@
+import listenbrainz_spark
+from listenbrainz_spark import config
+from pyspark.sql import Row, SparkSession
+from datetime import datetime
+import os, sys
+import train_models
+import recommend
+
+def prepare_user_data(table):
+    users_q = spark.sql("""
+            SELECT DISTINCT user_name
+             From %s
+        """ % (table))
+    users_rdd = users_q.rdd.zipWithIndex()
+    users_rdd = users_rdd.map(lambda user: Row(
+        user_id = user[1],
+        user_name = user[0][0],
+    ))
+    users_df = spark.createDataFrame(users_rdd)
+    return users_df, users_q
+
+def prepare_listen_data(table):
+    listens_df = spark.sql("""
+            SELECT listened_at
+                 , track_name
+                 , recording_msid
+                 , user_name
+             From %s
+        """ % (table))
+    return listens_df
+
+def prepare_recording_data(table):
+    recordings_q = spark.sql("""
+            SELECT track_name
+                 , recording_msid
+             From %s
+        """ % (table))
+    recordings_rdd = recordings_q.rdd.zipWithIndex()
+    recordings_rdd = recordings_rdd.map(lambda recording: Row(
+        recording_id = recording[1],
+        recording_msid = recording[0][1],
+        track_name = recording[0][0],
+    ))
+    recordings_df = spark.createDataFrame(recordings_rdd)
+    return recordings_df
+
+def get_playcounts_data(listens_df, users_df, recordings_df):
+
+    listens_df.createOrReplaceTempView('listen')
+    users_df.createOrReplaceTempView('user')
+    recordings_df.createOrReplaceTempView('recording')
+    playcounts_df = spark.sql("""
+        SELECT user_id,
```

Comment:
This is the place where we can do integer conversions -- this way we can skip several steps. Discussed in IRC, so I'll stop for now.